src/linker/implementation.py

```diff
@@ -77,8 +78,15 @@ def run(
             logger.info(f"Shutting down spark cluster for pipeline step ID {step_id}")
             session.control(job_id, drmaa.JobControlAction.TERMINATE)
 
-        for results_file in results_dir.glob("result.parquet"):
-            self.step.validate_output(results_file)
+        results_files = [file for file in results_dir.glob("result.parquet")]
```

Comment:
good looking out on this. I'm wondering if it would make sense to extract this into a step method. e.g. rename validate_output to validate_file or something and do
def validate_output(self, result_dir):
      ...
It's a bit more modular and eventually we should be able to know what the result directory and glob pattern just from the `step`, so everything is encapsulated.